### DISCUSSION TITLE
How to include javascript in snippet with enabled full page cache?

### DISCUSSION THREAD

**Discussion by storbahn** [pimcore/pimcore#7765] [Q&A / Support]
Is it possible to include javascript in a snippet with enabled full page cache? These snippets should be used on different pages.

We use the twig helper pimcore_head_script().appendFile() to include the javascript, but then only the rendered snippet template was cached and not the javascript. The output of pimcore_head_script is defined in the common layout file which almost all pages use.

**Comment by NiklasBr** [comment]
We are using the Twig function [render_esi(uri, options)](https://symfony.com/doc/current/reference/twig_reference.html#render-esi) to include our GDPR accept/decline snippet and it seems to work fine. Might work in your case as well.

**Reply by storbahn** [reply]
@NiklasBr thanks for the tip. but I think our CDN (AWS Cloudfront) is not supporting ESI
